### PR TITLE
variable: improve test coverage

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

This improves the test coverage of the variable package from 45.5% to 49.1%

### What is changed and how it works?

What's Changed:

Mainly tests. There is a small readability change in varsutil by using the new `HasGlobalScope()` function.

### Related changes

-None

### Check List 

Tests 

- Unit test

Side effects

- None

### Release note 

- No Release Note

### CODE REVIEW

File: sessionctx/variable/sysvar_test.go [hunk 1]
```diff
@@ -318,3 +322,227 @@ func (*testSysVarSuite) TestCollationServer(c *C) {
 	c.Assert(sv.SetSessionFromHook(vars, "utf8mb4_bin"), IsNil)
 	c.Assert(vars.systems[CharacterSetServer], Equals, "utf8mb4") // check it also changes charset.
 }
+
+func (*testSysVarSuite) TestTimeZone(c *C) {
+	sv := GetSysVar(TimeZone)
+	vars := NewSessionVars()
+
+	// TiDB uses the Golang TZ library, so TZs are case-sensitive.
+	// Unfortunately this is not strictly MySQL compatible. i.e.
```
Comment (wjhuang2016) [inline sessionctx/variable/sysvar_test.go:331]: Do we have an issue recording it?

File: sessionctx/variable/sysvar_test.go [hunk 2]
```diff
@@ -318,3 +322,227 @@ func (*testSysVarSuite) TestCollationServer(c *C) {
...
+	// This should not fail:
```
Comment (xhebox) [inline sessionctx/variable/sysvar_test.go:332]: I think we could add `TODO: ` marker, which is much easier to search, despite if there is an issue.

Comment (morgo) [reply sessionctx/variable/sysvar_test.go:332]: I've added a link to the issue directly. https://github.com/pingcap/tidb/issues/8087

**Review by xhebox — APPROVED** [review]

**Pull request merged**